vignettes/building_mocks.Rmd

```diff
@@ -0,0 +1,533 @@
+                     "{count} {percent}",
+                     count = frmt("xxx"),
+                     percent = frmt_when("==100"~ frmt(""),
+                                         "==0"~ "",
+                                         "TRUE" ~ frmt("(xx.x%)"))))
+  )) %>%
+  print_mock_gt(span_df)
+```
+
+To add a title, subtitle or footer:
+
+```{r warning=FALSE}
+mock_tbl_add_titles <- function(){
+  mock_tbl_adjust_cols() %>%
+    tfrmt(
+      # Specify title, subtitle, footer
+      title = "Table Name",
+      subtitle = "Study ID: GSK12345",
+      footer = "A footnote about stuff<br/>Here is a new line with more stuff")
+}
+  
+mock_tbl_add_titles() %>%
+  print_mock_gt(df)
+```
+
+## Making more complex mock tables
+
+Now let's make a slightly more complex table where we customise the value 
+formats by groups, labels and columns. Extra features such as titles, and 
+arranging column structure will be avoided to keep the code as simple as 
+possible.
+
+### Different value formatting between columns
+
+Here we have used the `crossing()` function twice with differing columns and 
+bound them using `bind_rows()`.
+
+```{r}
+df <- bind_rows(
+  crossing(label = c("label 1", "label 2", "label 3"),
+           column = c("T1", "T2", "PL"),
+           param = c("count", "percent")),
+  crossing(label = c("label 1", "label 2", "label 3"),
+           column = c("Risk Diff T1-PL", "Risk Diff T2-PL"),
+           param = c("num", "lower", "upper"))) %>%
+  arrange_all()
+
+df
+```
+
+This tfrmt specification is the same as the previous example but has one 
+additional `frmt_structure` which accounts for the value formatting we'd like 
+to apply to the the new columns.
+
+```{r}
+tfrmt(
+  # Specify columns in the data
+  label = "label",
+  param = "param",
+  column = "column",
+  values = value,
+  
+  # Specify body plan
+  body_plan = body_plan(
+    frmt_structure(group_val = ".default", label_val = ".default",
+                   frmt_combine(
+                     "{count} {percent}",
+                     count = frmt("xxx"),
+                     percent = frmt_when("==100"~ frmt(""),
+                                         "==0"~ "",
+                                         "TRUE" ~ frmt("(xx.x%)")))),
+    frmt_structure(group_val = ".default", label_val = ".default",
+                   frmt_combine(
+                     "{num} ({lower}, {upper})",
+                     num = frmt("xx.x"),
+                     lower = frmt_when("==100"~ frmt(""),
+                                       "==0"~ "",
+                                       "TRUE" ~ frmt("xx.x%")),
+                     upper = frmt_when("==100"~ frmt(""),
+                                       "==0"~ "",
+                                       "TRUE" ~ frmt("xx.x%"))))
+    ),
+  col_plan = col_plan(
+    -starts_with("ord"))
+  ) %>%
+  print_mock_gt(df)
+```
+
+Note that if you'd like to print a select few columns from your dataset you 
```

Comment:
Should we just show how this can be done in the col_plan instead?

Reply:
Changed example to demonstrate selecting specific columns with `col_plan` but also kept note mentioning that `filter()` can be used on `df` within `print_mock_gt()` as an alternative